extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -588,6 +605,16 @@ export class SchemaCompareResult {
 		});
 	}
 
+	private setButtonStatesForNoChanges(enableButtons: boolean): void {
+		// generate script and apply can only be enabled if the target is a database
+		if (this.targetEndpointInfo.endpointType === azdata.SchemaCompareEndpointType.Database) {
+			this.applyButton.enabled = enableButtons;
+			this.generateScriptButton.enabled = enableButtons;
+			this.applyButton.title = enableButtons ? applyEnabledMessage : applyNoChangesMessage;
+			this.generateScriptButton.title = enableButtons ? generateScriptEnabledMessage : generateScriptNoChangesMessage;
+		}
+	}
+
```

Comment:
don't remember the line .. but not sure why we need toggle while saving excludes state?

Reply:
In saveExcludeState()? If the newest exclusion results in no changes being currently included, the buttons should be disabled. Otherwise, they should be enabled if at least one change is included.